### ISSUE TITLE
Reputation should not drop below 2048, the "graduation" point

### ISSUE THREAD

**Issue by amihaiemil** [zerocracy/farm#848]
The reputation does not mean that you are a good developer at all -- if you think it means that, then the drop makes sense, because if you don't practice programming for a while, you forget, lose experience). And it doesn't mean you are a good developer or architect, since you can achieve it in many ways besides writing code or designing an application.

Instead, the reputation's only real purpose should be clear: it is the level of understanding of ZC's working model -- **in this regard, it should not drop bellow 2048**. Here are some ideas:

1. If they reached that point, they spent well over 2 months (best case, I believe), building it up, working in this fashion and understanding how it all works. You don't forget that even if you take a few months break, it's something that should have gotten in your blood if you reached it in the first place.

2. Inconsistencies: what happens if it drops below 2048? Do they lose their studends? Do they lose the ARC role in projects they already have? Are they assigned a new mentor? etc

3. It should, of course, drop from 10k until 2048; the difference from 2048 up shows which user is more active, faster etc. 

4. When do we kick them out if they can never reach -256? Well, we could kick them out if they have a long period of inactivity, say 1 year.

This is all in regards to the fact that ZC will most likely be a part-time thing, so once we graduate we shouldn't have the worry of the reputation anylonger + its purpose would be clearer.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by rexim** [comment]
The @yegor256's point was "it's bad to drop somebody from 4000 to a complete zero", which can happen in just 3 month right now. So we want to slow down that process a little bit. But it is difficult to control the dropping rate if all of the awards obey the 90 days rule. What if we make different kinds of awards live different amount of time? For example, award for `DEV` closed a bug could live longer than `REV` reviewed a PR or something like that. That makes controlling the reputation dropping rate more flexible.

**Comment by rexim** [comment]
Btw, with that mechanism we could "freeze" the awards earned before the "graduation".

**Comment by amihaiemil** [comment]
@rexim Given the simple purpose of the reputation (as I explained in the description), the whole mechanism you are describing is rather pointless... so far, users just need to learn the platform, period. Do you see it as a realistic representation for something else? I really don't.

**Comment by rexim** [comment]
@amihaiemil yes, absolutely correct. This is because I didn't take your interpretation of the purpose of the reputation into account when I wrote my proposal.

**Comment by t-izbassar** [comment]
@amihaiemil well, reputation is not only showing how well you know the platform. It shows efficiency of performers and how much values they bring to projects. That might be important for those who wants to hire zerocrat programmers to their projects. Also reputation affects the logic: with lower reputation you have reduced jobs limit, you can't apply to new non-sandbox projects and you can't have students and become an ARC in some project.

I mean that there are a lot of stuff attached to points so it is important to maintain that. I would say that there should be some checkpoints. For example you suggestion applicable not only for 2048 reputation. I would suggests those checkpoints:

- `256` points: if you reached that it means that you are good enough to join real projects. So being inactive in the platform should not affect that.
- `512` points
- `1024` points
- `2048` points

So those should be non removable checkpoints. Maybe we can attach some levels in zerocrat. For example in 0-256 you are `Level 1` performer. On 256-512 you are `Level 2` performer and so on. And this 90 days rule should not affect your level. You can maintain `Level 2` but 90 days rule will not get your points below 256.

**Comment by amihaiemil** [comment]
@t-izbassar 

> reputation is not only showing how well you know the platform. It shows efficiency of performers and how much values they bring to projects

Agreed, but that efficiency, performance and speed differenciation is represented by the difference from 2048 up. Graduation point is from where we should start drawing more conclusions related to efficiency; until graduation, it is assumed that they are still learning.

The rest of your comment I don't really understand, sorry. We already have tose milestones which would be respected if rep doesn't drop below 2048. 

And what is the point of introducing Levels that won't drop? If you have Level 4 (reached 2048 already in the past) and now you have 256 reputation, how is that helping? Rather than keeping track of who was where, through those Levels, isn't it much easier to just not go below 2048? Same result, simpler way.

**Comment by t-izbassar** [comment]
@amihaiemil well I suggested to expand you proposal. So you don't go below 256 if you already reached that point. You don't go below 512 and 1024 as well. Just to have more non reducible checkpoints. Level it's just a distinction, which might be taken in to consideration. So if you are Level 4 it means that your reputation 2048 or higher - it can't go below that point.

**Comment by amihaiemil** [comment]
@t-izbassar I see now, thanks. Yes, could also be an improvement.

**Comment by amihaiemil** [comment]
@yegor256 Are you suggesting to replace reputation with badges, once it reaches over 2048?

**Comment by yegor256** [comment]
@amihaiemil no. Once you reach 2048, you get a badge. That's all. You keep this badge on your account and it works for you when key decisions are made by the system.

**Comment by amihaiemil** [comment]
@yegor256 But I am a bit confused; will the reputation drop below 2048 once we have the badge? What is the purpose of the reputation in the system if we already have the badge and the system takes that into acoound? :)

**Comment by yegor256** [comment]
@amihaiemil yes, the reputation will go up and down by itself. But once you have a badge, it stays with you forever. When it's time to decide who can be a mentor, we select those who have "mentor" badge. We will use badges instead of reputation to make those decisions.

**Comment by amihaiemil** [comment]
@yegor256 I see, nice idea; but then reputation seems a bit useless; it would just give a short idea about how much and how fast the person is working on ZC, but I guess everyone will pay attention to the badges more

**Comment by yegor256** [comment]
@amihaiemil well, the reputation will matter for many other situations. But for some cases we will pay attention to the badges.

**Comment by yegor256** [comment]
@amihaiemil another idea here would be to increase 90 days to, say, 180 days.

**Comment by amihaiemil** [comment]
@yegor256 If reputation will keep going up and down, then it seems to me we are just artificially hiding the mentioned inconsistencies with labels -- if you want badge (I also like the idea), then I think it should be how @t-izbassar suggested above. Receiving a badge means you "leveld up" and your reputation will not go lower than that point. 

And whatever calculations 0crat does, it should do it only above the graduation point. So graduation point should be point 0 for whatever heuristics 0crat does; I imagine it's not going to calculate rate increases for people who haven't graduated.

**Comment by yegor256** [comment]
@amihaiemil we can't grant any reputation to anyone for a lifetime. You can earn reputation now, but stop working for a year. You come back in a year with 1024 points -- it's not correct. You have to earn them again. But a badge will help us understand how much you earned before.

**Comment by amihaiemil** [comment]
@yegor256 Yes, I agree with that idea, but I'm saying the reputation that should really matter in the system is only the one above the graduation point. You only graduate once and even if you stop working for a year, you don't forget how the model works and you can come back -- graduation is the one that shouldn't be erased. 

And the issue with going bellow graduation point (and reaching 0 eventually), even if we have the badge, is that I come back after a while, with 0 reputation and cannot refuse tasks right off the bat, since I will quickly reach -256 and be kicked out. If we can solve this problem, in any way, then I am happy :D

**Comment by amihaiemil** [comment]
@yegor256 on the other hand, truth is that it's rather hard to reach -256 anyway, since you're supposed to know the model already and at least you can find some bugs to improve the project and earn reputation... I guess we are fine with the badges then :-?

**Comment by yegor256** [comment]
@amihaiemil yes, let's go with the badges idea. Let's have three levels: "master" (when you reach 2048), "guru" (when it's 4096) and "champion" (when it's 16384).

BTW, if you work full time, 7 days a week, 4 hours per day, closing 8 tasks every day, your 90 reputation will be 21600.

**Comment by amihaiemil** [comment]
@yegor256 

>BTW, if you work full time, 7 days a week, 4 hours per day, closing 8 tasks every day, your 90 reputation will be 21600.

We both know that is the best case scenario. That won't be possible until there are many projects on the platform. Human factor is still high, we wait a lot for others :)

**Comment by amihaiemil** [comment]
@yegor256 but yeah, badges are fine. I'll edit the description here.

**Comment by 0crat** [comment]
Job [#848](https://github.com/zerocracy/farm/issues/848) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
The job [#848](https://github.com/zerocracy/farm/issues/848) assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1391543291); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)

**Comment by g4s8** [comment]
@yegor256 please see https://github.com/zerocracy/datum/issues/343

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by g4s8** [comment]
@0crat waiting for @yegor256 action in zerocracy/datum#343

**Comment by 0crat** [comment]
> @0crat waiting for @yegor256 action in zerocracy/datum#343 ([here](https://github.com/zerocracy/farm/issues/848#issuecomment-392311962))

@g4s8 The impediment for [#848](https://github.com/zerocracy/farm/issues/848) was registered successfully by @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by g4s8** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/848#issuecomment-472960368))

@g4s8 The user @g4s8[/z](https://www.0crat.com/u/g4s8) resigned from [#848](https://github.com/zerocracy/farm/issues/848), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
The job [#848](https://github.com/zerocracy/farm/issues/848) assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/016310be-245b-4169-a5bd-8c674054b195); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by carlosmiranda** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/848#issuecomment-541491969))

@carlosmiranda The user @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) resigned from [#848](https://github.com/zerocracy/farm/issues/848), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/848#issuecomment-541491969))

@carlosmiranda Job refused in 1 hours - no penalty, see [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
The job [#848](https://github.com/zerocracy/farm/issues/848) assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/7ad36be1-e177-4d56-93c5-78e4b4e8c6a9); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by carlosmiranda** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/848#issuecomment-541492955))

@carlosmiranda The user @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) resigned from [#848](https://github.com/zerocracy/farm/issues/848), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/848#issuecomment-541492955))

@carlosmiranda Job refused in 0 hours - no penalty, see [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/848#issuecomment-541492955))

@carlosmiranda There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@5be12c6b-b56e-4607-80d0-04b769beebe8, thread: pool-6-thread-7
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 1 error(s) in XML document: -1:-1: Duplicate unique value [gh:zerocracy/farm#848] declared for identity constraint of element "speed".
```

[0.49.19](https://github.com/zerocracy/farm): CID: [ad26e4d6-6ba6-49b9-9e7b-145ccd472d06](https://www.0crat.com/footprint/C3NDPUA8L/ad26e4d6-6ba6-49b9-9e7b-145ccd472d06), Type: "Order was canceled"
